InvenTree/users/models.py

```diff
@@ -1 +1,317 @@
+        'stock': [
+            'stock_stockitem',
+            'stock_stocklocation',
+            'stock_stockitemattachment',
+            'stock_stockitemtracking',
+            'stock_stockitemtestresult',
+        ],
+        'build': [
+            'part_part',
+            'part_partcategory',
+            'part_bomitem',
+            'build_build',
+            'build_builditem',
+            'stock_stockitem',
+            'stock_stocklocation',
+        ],
+        'purchase_order': [
+            'company_company',
+            'company_supplierpart',
+            'company_supplierpricebreak',
+            'order_purchaseorder',
+            'order_purchaseorderattachment',
+            'order_purchaseorderlineitem',
+        ],
+        'sales_order': [
+            'company_company',
+            'order_salesorder',
+            'order_salesorderattachment',
+            'order_salesorderlineitem',
+            'order_salesorderallocation',
+        ]
+    }
+
+    # Database models we ignore permission sets for
+    RULESET_IGNORE = [
+        # Core django models (not user configurable)
+        'admin_logentry',
+        'contenttypes_contenttype',
+        'sessions_session',
+
+        # Models which currently do not require permissions
+        'common_colortheme',
+        'common_currency',
+        'common_inventreesetting',
+        'company_contact',
+        'label_stockitemlabel',
+        'report_reportasset',
+        'report_testreport',
+        'part_partstar',
+    ]
+
+    RULE_OPTIONS = [
+        'can_view',
+        'can_add',
+        'can_change',
+        'can_delete',
+    ]
+
+    class Meta:
+        unique_together = (
+            ('name', 'group'),
+        )
+
+    name = models.CharField(
+        max_length=50,
+        choices=RULESET_CHOICES,
+        blank=False,
+        help_text=_('Permission set')
+    )
+
+    group = models.ForeignKey(
+        Group,
+        related_name='rule_sets',
+        blank=False, null=False,
+        on_delete=models.CASCADE,
+        help_text=_('Group'),
+    )
+
+    can_view = models.BooleanField(verbose_name=_('View'), default=True, help_text=_('Permission to view items'))
+
+    can_add = models.BooleanField(verbose_name=_('Create'), default=False, help_text=_('Permission to add items'))
+
+    can_change = models.BooleanField(verbose_name=_('Update'), default=False, help_text=_('Permissions to edit items'))
+
+    can_delete = models.BooleanField(verbose_name=_('Delete'), default=False, help_text=_('Permission to delete items'))
+    
+    @staticmethod
+    def get_model_permission_string(model, permission):
+        """
+        Construct the correctly formatted permission string,
+        given the app_model name, and the permission type.
+        """
+
+        app, model = model.split('_')
+
+        return "{app}.{perm}_{model}".format(
+            app=app,
+            perm=permission,
+            model=model
+        )
+
+    def __str__(self):
```

Comment:
Replace with something like?
```
    def __str__(self):
        """ Ruleset string representation """

        return f'{self.name} | v: {self.can_view} | a: {self.can_add} | ' \
               f'c: {self.can_change} | d: {self.can_delete}'
```

Reply:
Gasp, this actually is captured in the admin form... Can put this on backburner / for debug only